lib/index.js

```diff
@@ -4,25 +4,38 @@ const debug = require('debug')('cluster-harakiri');
 const WORKER_CONN_LIMIT = parseInt(process.env.HARAKIRI_WORKER_CONN_LIMIT, 10);
 const WORKER_TTL = parseInt(process.env.HARAKIRI_WORKER_TTL, 10);
 const WORKER_CHECK_INTERVAL = parseInt(process.env.HARAKIRI_WORKER_CHECK_INTERVAL, 10) || 30000;
+const WORKER_TERM_DELAY = parseInt(process.env.HARAKIRI_WORKER_TERM_DELAY, 10);
 
 const workers = {};
+const terminatingWorkers = [];
+let lastTermination = Date.now();
 let initialized = false;
 
 const setupHarakiri = (options) => {
   const ttl = (options && options.ttl) || WORKER_TTL;
   const connectionLimit = (options && options.connectionLimit) || WORKER_CONN_LIMIT;
   const restartWorker = !(options && options.restartWorker === false);
   const checkInterval = (options && options.checkInterval) || WORKER_CHECK_INTERVAL;
+  const termDelay = (options && options.termDelay) || WORKER_TERM_DELAY;
 
   const terminateWorker = (id) => {
+    lastTermination = Date.now();
     cluster.workers[id].disconnect();
-    delete workers[id];
-    debug(workers);
     if (restartWorker) {
       cluster.fork();
     }
   };
 
+  const queueTermination = (id) => {
+    delete workers[id];
```

Comment:
Do you want to delete this here or should it move to `terminateWorker`?

Reply:
It should be here. This is removing from our list of running workers. It is being added to the termination list below (if configured). If it is left in this list, it will continue to be evaluated for the connection limit and time limit even though it has already exceeded one of those